From 763a8c3b46217b51049dfbf058a678a5d5f2c8ba Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Wed, 21 Feb 2018 20:10:11 +0100
Subject: [PATCH 04/35] AdvancedDisplay: Fix strings

* It just so happens that not all devices will have the
  same options available, so allow them to specify what
  they have without breaking tranlations by moving the
  scenario names into individual strings from the arrays
* Update a few strings to be actually translatable
* Fix an off-by-one error for every display mode after
  "LineageOS (default)": UI is the same thing as the
  default mode, so remove it to make the mode selection
  actually choose the user-selected mode

Change-Id: Id5d659d79c10ea097c35b7416ef7c1cc93937756
---
 AdvancedDisplay/res/values/arrays.xml  | 27 +++++++++++++-------------
 AdvancedDisplay/res/values/strings.xml | 15 ++++++++++++++
 2 files changed, 28 insertions(+), 14 deletions(-)

diff --git a/AdvancedDisplay/res/values/arrays.xml b/AdvancedDisplay/res/values/arrays.xml
index a96bcd0..2e2eeb3 100644
--- a/AdvancedDisplay/res/values/arrays.xml
+++ b/AdvancedDisplay/res/values/arrays.xml
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!--
      Copyright (C) 2012-2014 The CyanogenMod Project
+     Copyright (C) 2018 The LineageOS Project
 
      Licensed under the Apache License, Version 2.0 (the "License");
      you may not use this file except in compliance with the License.
@@ -15,16 +16,15 @@
      limitations under the License.
 -->
 <resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
-    <string-array name="mdnie_scenario_entries">
-        <item>LineageOS (default)</item>
-        <item>UI</item>
-        <item>Video</item>
-        <item>Video warm</item>
-        <item>Video cold</item>
-        <item>Camera</item>
-        <item>Navigation</item>
-        <item>Gallery</item>
-        <item>VT</item>
+    <string-array name="mdnie_scenario_entries" translatable="false">
+        <item>@string/mdnie_scenario_ui</item>
+        <item>@string/mdnie_scenario_video</item>
+        <item>@string/mdnie_scenario_video_warm</item>
+        <item>@string/mdnie_scenario_video_cold</item>
+        <item>@string/mdnie_scenario_camera</item>
+        <item>@string/mdnie_scenario_navigation</item>
+        <item>@string/mdnie_scenario_gallery</item>
+        <item>@string/mdnie_scenario_vt</item>
     </string-array>
 
     <string-array name="mdnie_scenario_entries_values" translatable="false">
@@ -36,12 +36,11 @@
         <item>5</item>
         <item>6</item>
         <item>7</item>
-        <item>8</item>
     </string-array>
 
-    <string-array name="mdnie_negative_entries">
-        <item>Normal</item>
-        <item>Inverted</item>
+    <string-array name="mdnie_negative_entries" translatable="false">
+        <item>@string/mdnie_inverse_normal</item>
+        <item>@string/mdnie_inverse_inverse</item>
     </string-array>
 
     <string-array name="mdnie_negative_entries_values" translatable="false">
diff --git a/AdvancedDisplay/res/values/strings.xml b/AdvancedDisplay/res/values/strings.xml
index 89f55a9..99f7800 100644
--- a/AdvancedDisplay/res/values/strings.xml
+++ b/AdvancedDisplay/res/values/strings.xml
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!--
      Copyright (C) 2012-2014 The CyanogenMod Project
+     Copyright (C) 2018 The LineageOS Project
 
      Licensed under the Apache License, Version 2.0 (the "License");
      you may not use this file except in compliance with the License.
@@ -27,6 +28,20 @@
     <string name="mdnie_negative_title_head">Negative mode</string>
     <string name="mdnie_negative_summary_head">Enable/disable inverted colors</string>
 
+    <!-- Scenario names -->
+    <string name="mdnie_scenario_ui">LineageOS (default)</string>
+    <string name="mdnie_scenario_video">Video</string>
+    <string name="mdnie_scenario_video_warm">Video warm</string>
+    <string name="mdnie_scenario_video_cold">Video cold</string>
+    <string name="mdnie_scenario_camera">Camera</string>
+    <string name="mdnie_scenario_navigation">Navigation</string>
+    <string name="mdnie_scenario_gallery">Gallery</string>
+    <string name="mdnie_scenario_vt">Video call</string>
+
+    <!-- Inverse mode names -->
+    <string name="mdnie_inverse_normal">Normal</string>
+    <string name="mdnie_inverse_inverse">Inverted</string>
+
     <!-- DO NOT TRANSLATE Empty summary for dynamic preferences -->
     <string name="summary_empty" translatable="false"></string>
 </resources>
-- 
2.17.1

